common/bstrlib/bstrlib.h

```diff
@@ -29,6 +29,12 @@ extern "C" {
 # endif
 #endif
 
+#if defined(_MSC_VER)
+#define _ATTRIBUTE_FORMAT_(t,f,a)
```

Comment:
Formatted string is a GNU extension, that is supported by selected compilers, such as gcc or clang. When clang works in MSVC-compatible mode it will set `_MSC_VER` but will still provide the extension.

In general, I believe this code is still incorrect as it makes an exception for a known to be unsupported compiler, but does not make exceptions for unknown compilers that also are unsupported. You should not start the identifiers with underscore as well. Please change to:

```C
#if defined(__clang__) || defined(__GNUC__)
#define ATTRIBUTE_FORMAT_(t,f,a) __attribute__((format(t, f, a)))
#else
#define ATTRIBUTE_FORMAT_(t,f,a)
#endif
```

Reply:
Fixed in 582910fb96edefffa7c51442cc39681f90ece010